### PR TITLE
Refactored networks.py into networks module

### LINKED ISSUE #63: Refactor networks.py

### PR DESCRIPTION
**Pull request by cnellington**
Resolves #63

### CODE REVIEW

File: contextualized/networks/__init__.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+from contextualized.functions import *
+LINK_FUNCTIONS = {
+    'identity': linear_link_constructor(),
+    'logistic': logistic_constructor(),
+    'softmax': softmax_link_constructor()
+}
+
+from contextualized.regression.datasets import UnivariateDataset, MultitaskUnivariateDataset
```
Comment (blengerich) [inline contextualized/networks/__init__.py:14]: If these are used for both regression and networks, should they be in contextualized.regression?

File: contextualized/networks/__init__.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+DATASETS = {
+    "univariate": UnivariateDataset,
+    "multitask_univariate": MultitaskUnivariateDataset,
+}
+
+from contextualized.regression.losses import MSE
```
Comment (blengerich) [inline contextualized/networks/__init__.py:20]: If this is used for both regression and networks, should it be in contextualized.regression?

File: contextualized/networks/__init__.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+LOSSES = {
+    'mse': MSE
+}
+
+from contextualized.regression.regularizers import *
```
Comment (blengerich) [inline contextualized/networks/__init__.py:25]: If this is used for both regression and networks, should it be in contextualized.regression?

File: contextualized/networks/trainers.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import numpy as np
+
+from contextualized.regression.trainers import RegressionTrainer
+
+    
+class CorrelationTrainer(RegressionTrainer):
```
Comment (blengerich) [inline contextualized/networks/trainers.py:6]: Is correlation a network-specific function? Should we have a CorrelationTrainer in contextualized.regression.trainers?

File: contextualized/networks/lightning_modules.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    def __init__(self, context_dim, x_dim, **kwargs):
+        super().__init__(context_dim, x_dim, x_dim, **kwargs)
+    
+    def dataloader(self, C, X, **kwargs):
+        return super().dataloader(C, X, X, **kwargs)
+
+
+class MultitaskContextualizedCorrelation(TasksplitContextualizedUnivariateRegression):
+    """
+    Using umultitask nivariate contextualized regression to estimate Pearson's correlation
```
Comment (blengerich) [inline contextualized/networks/lightning_modules.py:30]: spelling

**Review by blengerich — CHANGES_REQUESTED** [review]
Some questions about placement in modules

**Comment by cnellington** [comment]
Correlation is a specific use case of univariate regression, so I agree it makes more sense to have a general correlation class / trainer in the regression module. I'll add what we have now to regression and make an issue to add the general ContextualizedCorrelation(C, X, Y) in the future.

**Review by blengerich — APPROVED** [review]
LGTM

**Pull request merged**